packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -138,6 +142,20 @@ sealed class _DebugSemanticsRoleChecks {
     });
     return error;
   }
+
+  static FlutterError? _semanticsCell(SemanticsNode node) {
+    if (node.parent?.role != SemanticsRole.table) {
```

Comment:
what if they are in a row?

Reply:
As I mentioned that in the PR description, I didn't add a role for table row in this PR,  because right now in framework tableRow is not a widget or a renderObject, but just some data.
I will update the tests if i add a role for table row later.